libp2p/protocols/secure/secure.nim

```diff
@@ -120,10 +120,23 @@ method readOnce*(s: SecureConn,
   doAssert(nbytes > 0, "nbytes must be positive integer")
 
   if s.buf.data().len() == 0:
-    let buf = await s.readMessage()
+    let (buf, err) = try:
+        (await s.readMessage(), nil)
+      except CatchableError as exc:
+        (@[], exc)
+
+    if err != nil:
```

Comment:
Lets not compare with `nil` directly because it can cause problems in future (Nim changes) so its better and recommended way is `isNil(err)`.